### PR TITLE
feat: Add support for stream_configs in FHIR store

### PR DESCRIPTION
**Pull request by jordan83**
Resource reference:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/healthcare_fhir_store

### CODE REVIEW

**Comment by xingao267** [comment]
@jordan83 few things you also need to do.

1) run `make generate_docs` to auto-update readme files.
2) run `make docker_test_lint` and `make docker_test_integration` to run the tests locally.

File: test/setup/main.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 module "project" {
   source  = "terraform-google-modules/project-factory/google"
-  version = "~> 3.0"
+  version = "~> 9.2.0"
```
Comment (xingao267) [inline test/setup/main.tf:19]: The tests failed due to a provider version conflict.

https://github.com/terraform-google-modules/terraform-google-healthcare/blob/master/test/setup/versions.tf needs to be updated. You can use 3.46.0 as https://github.com/terraform-google-modules/terraform-google-group/blob/main/test/setup/versions.tf

Comment (jordan83) [reply test/setup/main.tf:19]: Done.

File: test/setup/main.tf [hunk 2]
```diff
@@ -25,9 +25,18 @@ module "project" {
...
   activate_apis = [
-    "healthcare.googleapis.com",
     "iam.googleapis.com",
     "pubsub.googleapis.com",
     "serviceusage.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
+    "bigquery.googleapis.com"
   ]
+
+  activate_api_identities = [{
```
Comment (xingao267) [inline test/setup/main.tf:35]: why do we need this?

File: examples/simple_example/main.tf [hunk 3]
```diff
@@ -68,6 +74,14 @@ module "healthcare" {
       notification_config = {
         pubsub_topic = local.pubsub_topic
       }
+      stream_configs = [{
```
Comment (xingao267) [inline examples/simple_example/main.tf:77]: If there is gcloud commands available for stream configs, could you also follow the pattern in https://github.com/terraform-google-modules/terraform-google-healthcare/blob/master/test/integration/simple_example/controls/gcloud.rb to add some gcloud commands to check that the stream configs is successfully configured?

File: test/setup/main.tf (hunk 2 above)
Comment (jordan83) [reply test/setup/main.tf:35]: Healthcare api needs these roles in order to stream to BQ (which was added to the test example).

File: examples/simple_example/main.tf (hunk 3 above)
Comment (jordan83) [reply examples/simple_example/main.tf:77]: Definitely possible. That said, I think there are many gaps in the current test coverage. If this is different than checking for notification/pubsub configuration (which doesn't appear to be done currently) then I think it makes sense to add these checks in this PR. If not, I think we can leave it out.

Do you think we should file an issue to make more granular tests? The current tests don't check for permutations of optional resource configurations which I think is a bigger risk.

File: test/setup/main.tf (hunk 2 above)
Comment (jordan83) [reply test/setup/main.tf:35]: The healthcare SA needs this in order to enable BQ streaming.

File: examples/simple_example/main.tf (hunk 3 above)
Comment (xingao267) [reply examples/simple_example/main.tf:77]: There were some tests for notification configs before, but got removed in https://github.com/terraform-google-modules/terraform-google-healthcare/pull/16/files. Checking notification configs and stream configs should be pretty similar. You can leave them out for this PR but open an Issue to add those.

Comment (jordan83) [reply examples/simple_example/main.tf:77]: Opened: https://github.com/terraform-google-modules/terraform-google-healthcare/issues/23

File: main.tf [hunk 4]
```diff
@@ -57,6 +57,23 @@ resource "google_healthcare_fhir_store" "fhir_stores" {
       pubsub_topic = notification_config.value.pubsub_topic
     }
   }
+
+  dynamic "stream_configs" {
+    for_each = lookup(each.value, "stream_configs", [])
+
+    content {
+      resource_types = lookup(stream_configs.value, "resource_types", null)
```
Comment (xingao267) [inline main.tf:65]: Can you also add `resource_types` in the example just for demonstration purpose?

Comment (jordan83) [reply main.tf:65]: As discussed offline, there is also value in leaving resource_types out of the example as it tests that the "optional" aspect of the field is preserved.

**Review by umairidris — APPROVED** [review]

**Pull request merged**